packages/babel-parser/src/plugins/typescript/index.js

```diff
@@ -1644,39 +1643,43 @@ export default (superClass: Class<Parser>): Class<Parser> =>
           break;
 
         case "interface":
-          if (this.tsCheckLineTerminatorAndMatch(tt.name, next)) {
-            if (next) this.next();
+          if (this.tsCheckLineTerminator(next) && this.match(tt.name)) {
             return this.tsParseInterfaceDeclaration(node);
           }
           break;
 
         case "module":
-          if (next) this.next();
-          if (this.match(tt.string)) {
-            return this.tsParseAmbientExternalModuleDeclaration(node);
-          } else if (this.tsCheckLineTerminatorAndMatch(tt.name, next)) {
-            return this.tsParseModuleOrNamespaceDeclaration(node);
+          if (this.tsCheckLineTerminator(next)) {
+            if (this.match(tt.string)) {
+              return this.tsParseAmbientExternalModuleDeclaration(node);
+            } else if (this.match(tt.name)) {
+              return this.tsParseModuleOrNamespaceDeclaration(node);
+            }
           }
           break;
 
         case "namespace":
-          if (this.tsCheckLineTerminatorAndMatch(tt.name, next)) {
-            if (next) this.next();
+          if (this.tsCheckLineTerminator(next) && this.match(tt.name)) {
             return this.tsParseModuleOrNamespaceDeclaration(node);
           }
           break;
 
         case "type":
-          if (this.tsCheckLineTerminatorAndMatch(tt.name, next)) {
-            if (next) this.next();
+          if (this.tsCheckLineTerminator(next) && this.match(tt.name)) {
             return this.tsParseTypeAliasDeclaration(node);
           }
           break;
       }
     }
 
-    tsCheckLineTerminatorAndMatch(tokenType: TokenType, next: boolean) {
-      return (next || this.match(tokenType)) && !this.isLineTerminator();
+    tsCheckLineTerminator(next: boolean) {
+      if (next) {
+        if (this.hasFollowingLineBreak()) return false;
+        this.next();
+      } else if (this.isLineTerminator()) {
+        return false;
+      }
+      return true;
```

Comment:
Do you prefer?
```suggestion
    tsCheckLineTerminator(next: boolean) {
      if (next) {
        if (this.hasFollowingLineBreak()) return false;
        this.next();
        return true
      }
      return !this.isLineTerminator();
```

Reply:
I find them equally readable, feel free to do what you prefer.